### DISCUSSION TITLE
Transformers & Similarity

### DISCUSSION THREAD

**Discussion by filipematos95** [explosion/spaCy#9011] [Help: Coding & Implementations] [answered]
Hello, 

I am wondering what would be the best approach to compute the similarity of two sentences using transformer models. I have tried the built-in method of similarity but it has no vectors ( `UserWarning: [W008] Evaluating Doc.similarity based on empty vectors.)`

**Comment by adrianeboyd** [comment, marked as answer]
You can add custom vectors hooks to do this from the transformer tensors (it does require a bit of understanding due to the use of overlapping strided spans in the provided `trf` models), but I'd consider another alternative like sentence-transformers instead: https://www.sbert.net/
